client/src/components/Sub_ProductDetail/ImageGallery.jsx

```diff
@@ -6,11 +6,25 @@ function ImageGallery({ currentStyle, productID }) {
   // console.log('SHOW CURRENT STYLE: ', currentStyle);
   const [currentImageIndex, setCurrentImageIndex] = useState(0);
   const [styleMemo, setStyleMemo] = useState({});
+  const { style_id } = currentStyle;
 
   useEffect(() => {
     setCurrentImageIndex(0);
+    setStyleMemo({});
   }, [productID]);
 
+  useEffect(() => {
+    setStyleMemo({ ...styleMemo, [style_id]: currentImageIndex });
+  }, [currentImageIndex]);
+
+  useEffect(() => {
+    if (styleMemo[style_id]) {
+      setCurrentImageIndex(styleMemo[style_id]);
+    } else {
+      setCurrentImageIndex(0);
+    }
+  }, [currentStyle]);
```

Comment:
Am I correct in understanding these 3 useEffects are what makes your carousel work?

Reply:
actually these three useEffect is to reset the starting image index of the carousel when user switch to a different product or style. the styleMemo is to store the current image index of the current style so that when switching between styles, the index of the image currently selected should be maintained when the gallery updates for the new style. The actual implementation of carousel is in the subcomponent ImageSlide.